astropy/cosmology/core.py

```diff
@@ -1319,18 +1322,14 @@ def angular_diameter_distance_z1z2(self, z1, z2):
         if Ok0 < 0:
             raise CosmologyError('Ok0 must be >= 0 to use this method.')
 
-        outscalar = False
-        if not isiterable(z1) and not isiterable(z2):
-            outscalar = True
-
-        z1 = np.atleast_1d(z1)
-        z2 = np.atleast_1d(z2)
+        z1 = np.array(z1)
```

Comment:
If we really need this (see below), I'd put `np.asanyarray` (which doesn't make a copy and passes subclasses).